tests/sql/operators/test_load_file.py

```diff
@@ -715,3 +717,129 @@ def test_load_file_should_fail_loudly(sample_dag, invalid_path, caplog):
         test_utils.run_dag(sample_dag)
     expected_error = f"File(s) not found for path/pattern '{invalid_path}'"
     assert expected_error in caplog.text
+
+
+def is_dict_subset(superset: dict, subset: dict) -> bool:
+    """
+    Compare superset and subset to check if the latter is a subset of former.
+    Note: dict1 <= dict2 was not working on multilevel nested dicts.
+    """
+    for key, val in subset.items():
+        print(key, val)
+        if isinstance(val, dict):
+            if key not in superset:
+                return False
+            result = is_dict_subset(superset[key], subset[key])
+            if not result:
+                return False
+        elif superset[key] != val:
+            return False
+    return True
+
+
+@pytest.mark.parametrize(
+    "remote_files_fixture",
+    [{"provider": "google"}],
+    indirect=True,
+    ids=["google_gcs"],
+)
+@pytest.mark.parametrize(
+    "database_table_fixture",
+    [{"database": Database.BIGQUERY, "table": Table(conn_id="bigquery")}],
+    indirect=True,
+    ids=["bigquery"],
+)
+def test_aql_load_file_optimized_path_method_called(
+    sample_dag, database_table_fixture, remote_files_fixture
+):
+    """
+    Verify the correct method is getting called for specific source and destination.
+    """
+    db, test_table = database_table_fixture
+    file_uri = remote_files_fixture[0]
+
+    # (source, destination) : {
+    #   method_path: where source is file source path and destination is database
+    # and method_path is the path to method
+    #   expected_kwargs: subset of all the kwargs that are passed to method mentioned in the method_path
+    #   expected_args:  List of all the args that are passed to method mentioned in the method_path
+    # }
+    optimised_path_to_method = {
+        ("gs", "bigquery",): {
+            "method_path": "airflow.providers.google.cloud.hooks.bigquery.BigQueryHook.insert_job",
+            "expected_kwargs": {
+                "configuration": {
+                    "load": {
+                        "sourceUris": [file_uri],
+                        "destinationTable": {"tableId": test_table.name},
+                    }
+                }
+            },
+            "expected_args": (),
+        }
+    }
+
+    source = file_uri.split(":")[0]
+    destination = db.sql_type
+    mock_path = optimised_path_to_method[(source, destination)]["method_path"]
+    expected_kwargs = optimised_path_to_method[(source, destination)]["expected_kwargs"]
+    expected_args = optimised_path_to_method[(source, destination)]["expected_args"]
+
+    with mock.patch(mock_path) as method:
+        load_file(
+            input_file=File(file_uri),
+            output_table=test_table,
+        ).operator.execute({})
+        assert method.called
+        assert is_dict_subset(superset=method.call_args.kwargs, subset=expected_kwargs)
```

Comment:
Why are we checking for a subset and not an exact value?

Reply:
@because Matching complete dicts also entail we create the exact same dicts in the test cases that might have a certain logic to them and implementing them might be overkill for test case and as this test case is meant to be generic for all source-destination paths it becomes crucial for it to not have that logic. Instead of that, we can check for certain important args and kwargs and that's why we are using subset logic.

Reply:
@utkarsharma2 seems to make sense, could you give an example of a parameter we don't care about in the test?

Reply:
@tatiana - {'source_file': File(location="gs://dag-authoring/test/8e14a473-3f61-42d6-b903-f9daa22b75c50.csv",type="csv"), 'target_table': Table(conn_id='bigquery', name='_tmp_nomxb3e1i1m5tlxtt3rra55y6tjlap28gibfwgzed587wdo2', metadata=Metadata(schema='tmp_astro', database='astronomer-dag-authoring'), columns=[], temp=True), 'if_exists': 'replace'}